src/jit/valuenum.cpp

```diff
@@ -3163,6 +3163,51 @@ void ValueNumStore::GetConstantBoundInfo(ValueNum vn, ConstantBoundInfo* info)
     }
 }
 
+//------------------------------------------------------------------------
+// IsVNArrLenUnsignedBound: Checks if the specified vn represents an
+//    expression such as "(uint)i < (uint)a.len" that imply that the
+//    array index is valid (0 <= i && i < a.len).
+//
+// Arguments:
+//    vn - Value number to query
+//    info - Pointer to a ArrLenUnsignedBoundInfo object to return
+//           information about the expression. Not populated if the
+//           vn expression isn't suitable (e.g. i <= a.len)
+
+bool ValueNumStore::IsVNArrLenUnsignedBound(ValueNum vn, ArrLenUnsignedBoundInfo* info)
```

Comment:
I went back and forth on this one partly because of this inconsistency. An early version simply returned just `bool` like the others but that isn't enough to create the assertion. Another version returned `VNFuncApp` which the assertion creation code had to parse but it didn't feel quite right to have non VN code do that. So I ended up with a compromise and combined `Is` and `GetInfo` functions into one.